oms-azure-storage-analytics-solution/scripts/AzureSAIngestionLogs-MS-Mgmt-SA.ps1

```diff
@@ -1067,221 +1020,224 @@ $Starttimer = get-date
 
 
 
-$colParamsforChild|foreach {
 
-    $splitmetrics = $null
-    $splitmetrics = $_
-    $Job = [powershell]::Create().AddScript($ScriptBlock).AddArgument($hash).AddArgument($splitmetrics).Addargument($i)
-    $Job.RunspacePool = $RunspacePool
-    $Jobs += New-Object PSObject -Property @{
-        RunNum = $i
-        Pipe   = $Job
-        Result = $Job.BeginInvoke()
-
-    }
-	
-    $i++
-}
+   
 
-write-output  "$(get-date)  , started $i Runspaces "
-Write-Output "After dispatching runspaces $([System.gc]::gettotalmemory('forcefullcollection') /1MB) MB"
-$jobsClone = $jobs.clone()
-Write-Output "Waiting.."
+foreach ($sa in @($colParamsforChild)) {
 
 
 
-$s = 1
-Do {
 
-    Write-Output "  $(@($jobs.result.iscompleted|where{$_  -match 'False'}).count)  jobs remaining"
+    $prikey = $storageaccount = $rg = $type = $null
+    $storageaccount = $sa.Split(';')[0]
+    $rg = $sa.Split(';')[1]
+    $type = $sa.Split(';')[2]
+    $tier = $sa.Split(';')[3]
+    $kind = $sa.Split(';')[4]
 
-    foreach ($jobobj in $JobsClone) {
 
-        if ($Jobobj.result.IsCompleted -eq $true) {
-            $jobobj.Pipe.Endinvoke($jobobj.Result)
-            $jobobj.pipe.dispose()
-            $jobs.Remove($jobobj)
-        }
-    }
+    If ($type -eq 'ARM') {
+        $Uri = "https://management.azure.com/subscriptions/{3}/resourceGroups/{2}/providers/Microsoft.Storage/storageAccounts/{1}/listKeys?api-version={0}" -f $ApiVerSaArm, $storageaccount, $rg, $SubscriptionId 
+        $keyresp = Invoke-WebRequest -Uri $uri -Method POST  -Headers $headers -UseBasicParsing
+        $keys = ConvertFrom-Json -InputObject $keyresp.Content
+        $prikey = $keys.keys[0].value
 
 
-    IF ($([System.gc]::gettotalmemory('forcefullcollection') / 1MB) -gt 200) {
-        [gc]::Collect()
     }
+    Elseif ($type -eq 'Classic') {
+        $Uri = "https://management.azure.com/subscriptions/{3}/resourceGroups/{2}/providers/Microsoft.ClassicStorage/storageAccounts/{1}/listKeys?api-version={0}" -f $ApiVerSaAsm, $storageaccount, $rg, $SubscriptionId 
```

Comment:
let's make these cloud agnostic and pass the keys into the script from the template, so we don't have to hard-code the management endpoint